imageio/imageio-tga/src/main/java/com/twelvemonkeys/imageio/plugins/tga/TGAImageReader.java

```diff
@@ -196,8 +196,15 @@ public BufferedImage read(final int imageIndex, final ImageReadParam param) thro
                     readRowByte(input, height, srcRegion, header.getOrigin(), xSub, ySub, rowDataByte, destRaster, clippedRow, y);
                     break;
                 case 16:
-                    short[] rowDataUShort = ((DataBufferUShort) rowRaster.getDataBuffer()).getData();
-                    readRowUShort(input, height, srcRegion, header.getOrigin(), xSub, ySub, rowDataUShort, destRaster, clippedRow, y);
+                    final DataBuffer dataBuffer = rowRaster.getDataBuffer();
+                    if (dataBuffer instanceof DataBufferByte) {
+                        byte[] rowDataByte16 = ((DataBufferByte) rowRaster.getDataBuffer()).getData();
+                        readRowByte(input, height, srcRegion, header.getOrigin(), xSub, ySub, rowDataByte16, destRaster, clippedRow, y);
+                    } else {
+                        // DataBufferUShort and any other
+                        short[] rowDataUShort = ((DataBufferUShort) rowRaster.getDataBuffer()).getData();
+                        readRowUShort(input, height, srcRegion, header.getOrigin(), xSub, ySub, rowDataUShort, destRaster, clippedRow, y);
+                    }
```

Comment:
Good catch! There is certainly a bug in the current `TGAImageReader` concerning 16 bit unsigned gray.

The suggested fix, however, doesn't properly read the 16 bit values. 

Instead, we can change the `getRawImageType`, to handle the fact that there is such a thing as 16 bit gray samples.

Line 119 and on:

```java
            case TGA.IMAGETYPE_MONOCHROME:
            case TGA.IMAGETYPE_MONOCHROME_RLE:
                switch (header.getPixelDepth()) {
                    case 8:
                        return ImageTypeSpecifiers.createFromBufferedImageType(BufferedImage.TYPE_BYTE_GRAY);
                    case 16:
                        return ImageTypeSpecifiers.createFromBufferedImageType(BufferedImage.TYPE_USHORT_GRAY);
                    default:
                        throw new IIOException("Unknown pixel depth for monochrome: " + header.getPixelDepth());
                }
```